backend/press/press.go

```diff
@@ -0,0 +1,1376 @@
+			}
+		}
+	} else {
+		// Function that updates object
+		update := func(ctx context.Context, in io.Reader, src fs.ObjectInfo, options ...fs.OpenOption) (fs.Object, error) {
+			return o.Object, o.Object.Update(ctx, in, src, options...)
+		}
+		// If we are, just update the object and metadata
+		newObject, err = o.f.putWithCustomFunctions(ctx, in, src, options, update, updateMeta, compressible, mimeType, true)
+	}
+	if err != nil {
+		return err
+	}
+	// Update object metadata and return
+	o.Object = newObject.Object
+	o.meta = newObject.meta
+	o.size = newObject.size
+	return nil
+}
+
+// This will initialize the variables of a new press Object. The metadata object, mo, and metadata struct, meta, must be specified.
+func (f *Fs) newObject(o fs.Object, mo fs.Object, meta *ObjectMetadata) *Object {
+	return &Object{
+		Object: o,
+		f:      f,
+		mo:     mo,
+		moName: mo.Remote(),
+		size:   meta.Size,
+		meta:   meta,
+	}
+}
+
+// This initializes the variables of a press Object with only the size. The metadata will be loaded later on demand.
+func (f *Fs) newObjectSizeAndNameOnly(o fs.Object, moName string, size int64) *Object {
+	return &Object{
+		Object: o,
+		f:      f,
+		mo:     nil,
+		moName: moName,
+		size:   size,
+		meta:   nil,
+	}
+}
+
+// This loads the metadata of a press Object if it's not loaded yet
+func (o *Object) loadMetadataIfNotLoaded(ctx context.Context) (err error) {
+	err = o.loadMetadataObjectIfNotLoaded(ctx)
+	if err != nil {
+		return err
+	}
+	if o.meta == nil {
+		o.meta = readMetadata(ctx, o.mo)
+	}
+	return err
+}
+
+// This loads the metadata object of a press Object if it's not loaded yet
+func (o *Object) loadMetadataObjectIfNotLoaded(ctx context.Context) (err error) {
+	if o.mo == nil {
+		o.mo, err = o.f.Fs.NewObject(ctx, o.moName)
+	}
+	return err
+}
+
+// Fs returns read only access to the Fs that this object is part of
+func (o *Object) Fs() fs.Info {
+	return o.f
+}
+
+// Return a string version
+func (o *Object) String() string {
+	if o == nil {
+		return "<nil>"
+	}
+	return o.Remote()
+}
+
+// Remote returns the remote path
+func (o *Object) Remote() string {
+	origFileName, _, _, err := processFileName(o.Object.Remote())
+	if err != nil {
+		fs.Errorf(o, "Could not get remote path for: %s", o.Object.Remote())
+		return o.Object.Remote()
+	}
+	return origFileName
+}
+
+// Size returns the size of the file
+func (o *Object) Size() int64 {
+	if o.meta == nil {
+		return o.size
+	}
+	return o.meta.Size
+}
+
+// MimeType returns the MIME type of the file
+func (o *Object) MimeType(ctx context.Context) string {
+	err := o.loadMetadataIfNotLoaded(ctx)
+	if err != nil {
+		return "error/error"
+	}
+	return o.meta.MimeType
+}
+
+// Hash returns the selected checksum of the file
+// If no checksum is available it returns ""
+func (o *Object) Hash(ctx context.Context, ht hash.Type) (string, error) {
+	err := o.loadMetadataIfNotLoaded(ctx)
+	if err != nil {
+		return "", err
+	}
+	if ht&hash.MD5 == 0 {
```

Comment:
`ht hash.Type` isn't used as bitmap here so usually `ht == hash.MD5` is what we use.

You can put this first before the `loadMetadataIfNotLoaded` if you want to save loading metadata when we couldn't return a hash.